### DISCUSSION TITLE
Keycloak Local Build failing due to dependencies issue

### DISCUSSION THREAD

**Discussion by gjain-sag** [keycloak/keycloak#12159] [Q&A] [answered]
Hi Team,

I was trying to build keycloak-17.0.1 release server locally using command
 ` mvn -Pdistribution -pl distribution/server-dist -am -Dmaven.test.skip clean install`

but build is failing with below error. Any idea what could be wrong here.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 17.0.1:
[INFO]
[INFO] Keycloak ........................................... SUCCESS [  6.747 s]
[INFO] Keycloak Common .................................... SUCCESS [  8.298 s]
[INFO] Keycloak Core ...................................... SUCCESS [  8.488 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  7.090 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [ 14.909 s]
[INFO] Keycloak Kerberos Federation ....................... SUCCESS [  1.127 s]
[INFO] Keycloak LDAP UserStoreProvider .................... SUCCESS [  3.402 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  8.537 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  6.202 s]
[INFO] Keycloak REST Services ............................. SUCCESS [ 35.868 s]
[INFO] Keycloak JS Integration ............................ SUCCESS [ 20.291 s]
[INFO] Keycloak Themes .................................... SUCCESS [03:27 min]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.038 s]
[INFO] Keycloak Model JPA ................................. SUCCESS [  8.571 s]
[INFO] Keycloak Model Java Annotations and Processor ...... SUCCESS [  1.009 s]
[INFO] Keycloak Model Map ................................. SUCCESS [  8.798 s]
[INFO] Keycloak Model Infinispan .......................... SUCCESS [ 13.019 s]
[INFO] Keycloak Model Hot Rod ............................. SUCCESS [  4.646 s]
[INFO] Keycloak SSSD Federation ........................... SUCCESS [  4.168 s]
[INFO] KeyCloak Authz: Parent ............................. SUCCESS [  0.047 s]
[INFO] KeyCloak AuthZ: Provider Parent .................... SUCCESS [  0.047 s]
[INFO] KeyCloak AuthZ: Common Policy Providers ............ SUCCESS [  1.418 s]
[INFO] Keycloak WildFly Integration ....................... SUCCESS [  0.036 s]
[INFO] Keycloak WildFly Add User Script ................... SUCCESS [  3.951 s]
[INFO] Keycloak WildFly Extensions ........................ SUCCESS [  3.310 s]
[INFO] Keycloak WildFly Server Subsystem .................. SUCCESS [  9.261 s]
[INFO] Keycloak Integration ............................... SUCCESS [  0.036 s]
[INFO] Keycloak Client CLI ................................ SUCCESS [  0.035 s]
[INFO] Keycloak Client Registration CLI ................... SUCCESS [  4.047 s]
[INFO] Keycloak Admin CLI ................................. SUCCESS [  3.125 s]
[INFO] Keycloak Client CLI Distribution ................... SUCCESS [  1.588 s]
[INFO] Distribution Parent ................................ SUCCESS [  0.032 s]
[INFO] Feature Pack Builds ................................ SUCCESS [  0.034 s]
[INFO] Keycloak Feature Pack: Server Dependencies ......... SUCCESS [  0.052 s]
[INFO] Keycloak Feature Pack: Server ...................... SUCCESS [ 34.419 s]
[INFO] Galleon Feature Pack Builds ........................ SUCCESS [  0.109 s]
[INFO] Keycloak Galleon Feature Pack: Server .............. FAILURE [ 17.503 s]
[INFO] Keycloak Server Legacy WildFly Distribution ........ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  07:33 min
[INFO] Finished at: 2022-05-24T07:16:46+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project keycloak-server-galleon-pack: Could not resolve dependencies for project org.keycloak:keycloak-server-galleon-pack:pom:17.0.1: Failed to collect dependencies at org.wildfly:wildfly-ee-feature-pack-pruned:pom:26.0.1.Final -> org.jboss.activemq.artemis.integration:artemis-wildfly-integration:jar:1.0.5: Failed to read artifact descriptor for org.jboss.activemq.artemis.integration:artemis-wildfly-integration:jar:1.0.5: Could not transfer artifact org.jboss.activemq.artemis.integration:artemis-wildfly-integration:pom:1.0.5 from/to jboss (https://repository.jboss.org/nexus/content/groups/public/): Transfer failed for https://repository.jboss.org/nexus/content/groups/public/org/jboss/activemq/artemis/integration/artemis-wildfly-integration/1.0.5/artemis-wildfly-integration-1.0.5.pom 503 Service Unavailable -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-server-galleon-pack

**Comment by andreaTP** [comment]
Hi @gjain-sag , you are right and I can reproduce:
https://github.com/andreaTP/keycloak/runs/6570317132?check_suite_focus=true#step:5:13648

Possibly that's just a temporary outage.

**Comment by gjain-sag** [comment]
Hi @andreaTP ,

yes outage could be one issue , also some artifacts are getting resolved from maven repository :
[INFO] --- wildfly-galleon-maven-plugin:5.2.7.Final:build-feature-pack (keycloak-server-galleon-pack-build) @ keycloak-server-galleon-pack ---
Downloading from jboss-enterprise-maven-repository: https://maven.repository.redhat.com/ga/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.pom
Downloading from jboss: https://repository.jboss.org/nexus/content/groups/public/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.pom (0 B at 0 B/s)
Downloading from jboss: https://repository.jboss.org/nexus/content/groups/public/org/jboss/hal/hal-bom/3.4.4.Final/hal-bom-3.4.4.Final.pom
Downloading from jboss-nexus: https://repository.jboss.org/nexus/content/groups/public/org/jboss/hal/hal-bom/3.4.4.Final/hal-bom-3.4.4.Final.pom

but the dependency which failed is no longer available in maven repository, like the one `hal-bom-3.4.4.Final.pom`. Can this be a cause too?

**Reply by andreaTP** [reply]
@gjain-sag I can confirm that `repository.jboss.org` is down. The issue has been reported, not much we can do other than waiting for a solution.

**Comment by andreaTP** [comment, marked as answer]
Forgot to update but the `repository.jboss.org` outage has been resolved and everything is running fine for some hours now.

**Comment by aledeulo** [comment]
Hello, I have the same issue. I checked out from main branch, and also form the latest release tag, and I'm having the same issue but with other dependencies.

```
[INFO] Reactor Summary for Keycloak 999.0.0-SNAPSHOT:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [  0.358 s]
[INFO] Keycloak BOM for adapters .......................... SUCCESS [  0.011 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [  0.009 s]
[INFO] Keycloak BOM utilities for the quickstarts ......... SUCCESS [  0.012 s]
[INFO] Keycloak ........................................... SUCCESS [  0.027 s]
[INFO] Keycloak Common .................................... SUCCESS [  1.494 s]
[INFO] Keycloak Core ...................................... SUCCESS [  1.399 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  0.976 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  1.644 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.012 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  0.129 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.565 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  1.112 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.012 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.069 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.719 s]
[INFO] Keycloak REST Services ............................. SUCCESS [  0.765 s]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SUCCESS [  0.073 s]
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SUCCESS [  0.179 s]
[INFO] Keycloak JavaScript Adapter (JAR) .................. FAILURE [  0.527 s]
[INFO] Keycloak JavaScript Parent ......................... SKIPPED
[INFO] Keycloak Admin UI .................................. SKIPPED
[INFO] Keycloak Account UI ................................ SKIPPED
[INFO] Keycloak Themes .................................... SKIPPED
[INFO] Keycloak Dependencies Parent ....................... SKIPPED
[INFO] Keycloak Dependencies Server Min ................... SKIPPED
[INFO] Keycloak Model JPA ................................. SKIPPED
[INFO] Keycloak Kerberos Federation ....................... SKIPPED
[INFO] Keycloak LDAP UserStoreProvider .................... SKIPPED
[INFO] Keycloak Database Support - REST services .......... SKIPPED
[INFO] Keycloak Model Infinispan .......................... SKIPPED
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Dependencies Server All ................... SKIPPED
[INFO] Keycloak Quarkus Parent ............................ SKIPPED
[INFO] Keycloak Configuration API ......................... SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Docs Parent ............................... SKIPPED
[INFO] Keycloak Guides Maven Plugin ....................... SKIPPED
[INFO] Keycloak Guides .................................... SKIPPED
[INFO] Keycloak Federation ................................ SKIPPED
[INFO] Keycloak Util Embedded LDAP ........................ SKIPPED
[INFO] Keycloak Util Parent ............................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Admin REST Client JavaEE For Tests ........ SKIPPED
[INFO] Keycloak Admin REST Client for tests ............... SKIPPED
[INFO] Keycloak Client Registration API ................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Adapter SPI ............................... SKIPPED
[INFO] Keycloak Undertow Integration SPI .................. SKIPPED
[INFO] Common JBoss/Wildfly Core Classes .................. SKIPPED
[INFO] Keycloak Client Adapter SPI Modules ................ SKIPPED
[INFO] Keycloak SAML Client Adapter Public API ............ SKIPPED
[INFO] Keycloak SAML Client Adapter Core .................. SKIPPED
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SKIPPED
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SKIPPED
[INFO] Keycloak SAML Wildfly Integration .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Modules ............... SKIPPED
[INFO] Keycloak OIDC Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Adapters .................................. SKIPPED
[INFO] Keycloak Authz: Client API ......................... SKIPPED
[INFO] Keycloak Authz: Policy Enforcer .................... SKIPPED
[INFO] Keycloak Misc ...................................... SKIPPED
[INFO] keycloak-test-helper ............................... SKIPPED
[INFO] Keycloak JavaScript Admin Client ................... SKIPPED
[INFO] Keycloak UI shared components ...................... SKIPPED
[INFO] Keycloak JavaScript Adapter ........................ SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.503 s
[INFO] Finished at: 2024-07-05T11:52:15+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.15.0:install-node-and-pnpm (default) on project keycloak-js-adapter-jar: Could not download Node.js: Could not download https://nodejs.org/dist/v20.13.0/node-v20.13.0-darwin-arm64.tar.gz: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-js-adapter-jar
```
